pkg/proc/interface.go

```diff
@@ -102,8 +102,6 @@ type ProcessManipulation interface {
 
 // BreakpointManipulation is an interface for managing breakpoints.
 type BreakpointManipulation interface {
-	Breakpoints() *BreakpointMap
-	SetBreakpoint(addr uint64, kind BreakpointKind, cond ast.Expr) (*Breakpoint, error)
-	ClearBreakpoint(addr uint64) (*Breakpoint, error)
-	ClearInternalBreakpoints() error
+	WriteBreakpointFn(addr uint64) (string, int, *Function, []byte, error)
```

Comment:
I'm going to insist that these internal functions are moved to an interface other than proc.Process that we don't embed inside Target. We had two "internal methods users of proc aren't supposed to call" before and now we have four.

Reply:
I'm fine with that as long as we agree it's temporary. I think in the end all backends should be in `pkg/proc/internal/...` so they are completely unexposed to anything outside. Everything should use `Target` once this refactor is done, and the backends should not leak out as they simply aren't safe to use by just anything.